llvm/lib/Transforms/IPO/GlobalOpt.cpp

```diff
@@ -2404,6 +2405,42 @@ static bool OptimizeEmptyGlobalCXXDtors(Function *CXAAtExitFn) {
   return Changed;
 }
 
+static Function *hasSideeffectFreeStaticResolution(GlobalIFunc &IF) {
+  Function *Resolver = IF.getResolverFunction();
+  if (!Resolver)
+    return nullptr;
```

Comment:
You need to guard against interposable linkage here, e.g. `weak`. Unless this is forbidden for ifuncs?